### PR TITLE
Work around removal of imp in Python 3.12.

### PR DESCRIPTION
**Pull request by limburgher**

### CODE REVIEW

**Comment by limburgher** [comment]
See https://docs.python.org/3.12/whatsnew/3.12.html#removed

File: docs/conf.py [hunk 1]
```diff
@@ -1,12 +1,26 @@
...
+    # sys.modules[module.__name__] = module
+    loader.exec_module(module)
+    return module
+
+
 # Add the extensions folder...
 sys.path.insert(0, os.path.abspath("./_extensions"))
 
-_version = imp.load_source("setup", "../src/klein/_version.py")
+_version = load_source("setup", "../src/klein/_version.py")
```
Comment (adiroiban) [inline docs/conf.py:23]: I guess that `load_source` this is also a private / helper

```suggestion
_version = _load_source("setup", "../src/klein/_version.py")
```

I guess that this is ok and loading and executing the code works.

there is also the option of using `importlib.metadata.version`

```
project = "Klein"
version = importlib.metadata.version(project)
```

**Review by adiroiban — APPROVED** [review]
Many thanks for the changes.

This looks good and it works.

On https://klein--635.org.readthedocs.build/en/635/ I can see 

```
  <title>Klein, a Web Micro-Framework &mdash; Klein 23.5.0.rc1 documentation</title>
```

so the version was correctly loaded.

I guess that as long as it works, and this is only about building the documentation, this can be merged.

`importlib.metadata.version` would require the klein package to be installed at the time the documentatin is built.

I am adding the other devs back in review loop to see if the prefered way for getting a project version is `importlib.metadata.version`

File: docs/conf.py (hunk 1 above)
Comment (glyph) [reply docs/conf.py:23]: Given that this is happening in a configuration file and not within Klein's API surface, I think skipping the underscore is fine.

**Pull request merged**